paseos/attitude/attitude_model.py

```diff
@@ -0,0 +1,295 @@
+        actor_initial_attitude_in_rad: list[float] = [0., 0., 0.],
+        actor_initial_angular_velocity: list[float] = [0., 0., 0.],
+        # pointing vector in body frame: (defaults to body z-axis)
+        actor_pointing_vector_body: list[float] = [0., 0., 1.],
+        actor_residual_magnetic_field: list[float] = [0., 0., 0.],
+    ):
+        """Creates an attitude model to model actor attitude based on
+        initial conditions (initial attitude and angular velocity) and
+        external disturbance torques.
+
+        Args:
+            actor (SpacecraftActor): Actor to model.
+            actor_initial_attitude_in_rad (list of floats): Actor's initial attitude ([roll, pitch, yaw]) angles.
+                Defaults to [0., 0., 0.].
+            actor_initial_angular_velocity (list of floats): Actor's initial angular velocity vector.
+                Defaults to [0., 0., 0.].
+            actor_pointing_vector_body (list of floats): User defined vector in the Actor body. Defaults to [0., 0., 1]
+            actor_residual_magnetic_field (list of floats): Actor's own magnetic field modeled as dipole moment vector.
+                Defaults to [0., 0., 0.].
+        """
+        self._actor = local_actor
+        # convert to np.ndarray
+        self._actor_attitude_in_rad = np.array(actor_initial_attitude_in_rad)
+        self._actor_angular_velocity = np.array(actor_initial_angular_velocity)
+
+        # normalize inputted pointing vector & convert to np.ndarray
+        self._actor_pointing_vector_body = np.array(
+            actor_pointing_vector_body
+        ) / np.linalg.norm(np.array(actor_pointing_vector_body))
+
+        # pointing vector expressed in Earth-centered inertial frame
+        self._actor_pointing_vector_eci = rpy_to_eci(
+            body_to_rpy(self._actor_pointing_vector_body, self._actor_attitude_in_rad),
+            np.array(self._actor.get_position(self._actor.local_time)),
+            np.array(self._actor.get_position_velocity(self._actor.local_time)[1]),
+        )
+
+        # angular velocity vector expressed in Earth-centered inertial frame
+        self._actor_angular_velocity_eci = rpy_to_eci(
+            body_to_rpy(self._actor_angular_velocity, self._actor_attitude_in_rad),
+            np.array(self._actor.get_position(self._actor.local_time)),
+            np.array(self._actor.get_position_velocity(self._actor.local_time)[1]),
+        )
+
+        # actor residual magnetic field (modeled as dipole)
+        self._actor_residual_magnetic_field = np.array(actor_residual_magnetic_field)
+
+    def _nadir_vector(self):
+        """Compute unit vector pointing towards earth, in the inertial frame.
+
+        Returns:
+            np array ([x, y, z]): unit nadir vector in ECIF (Earth-centered inertial frame)
+        """
+        u = np.array(self._actor.get_position(self._actor.local_time))
+        return -u / np.linalg.norm(u)
+
+    def _earth_magnetic_dipole_moment(self):
```

Comment:
That is true, I put it in central_body. I put a todo to implement other central bodies instead of Earth, I can remove that todo and add an issue later (or don't do any). Is this okay?